ord_schema/visualization/template.html

```diff
@@ -14,101 +14,153 @@
+    {# FOURTH ROW = compound amount #}
     <tr>
-        {% for key,input in reaction.inputs|sort_addition_order %}
-            {% for compound in input.components %}
-                <td>{{ compound|compound_amount }}</td>
+        {% for key, input in reaction.inputs|sort_addition_order %}
+            {% for compound, border in input.components|get_input_borders %}
+                <td class="{{ border }}">{{ compound|compound_amount }}</td>
             {% endfor %}
         {% endfor %}
-        <td></td>
+        <td class="clean"></td>
         {% for outcome in reaction.outcomes %}
             {% for product in outcome.products %}
-                <td>
+                <td class="clean">
                     {% if product.compound_yield.value %}
-                    {{ product.compound_yield.value|round(3) }}%
-                    {% if product.compound_yield.precision %}
-                        (p/m {{ product.compound_yield.precision|round(3) }}%)
-                    {% endif %}
-                    yield
-                    {% if product.analysis_yield %}
-                        by 
-                        {% for key in product.analysis_yield %}
-                            {% if loop.last and loop.index > 1 %} and {% endif %}
-                            {{ outcome.analyses[key]|analysis_format }}
-                            {% if not loop.last %}, {% endif %}
-                        {% endfor %}
+                        {{ product.compound_yield.value|round(3) }}%
+                        {% if product.compound_yield.precision %}
+                            (p/m {{ product.compound_yield.precision|round(3) }}%)
+                        {% endif %}
+                        yield
+                        {% if product.analysis_yield %}
+                            by
+                            {% for key in product.analysis_yield %}
+                                {% if loop.last and loop.index > 1 %} and {% endif %}
+                                {{ outcome.analyses[key]|analysis_format }}
+                                {% if not loop.last %}, {% endif %}
+                            {% endfor %}
+                        {% endif %}
                     {% endif %}
-                {% endif %}
                 </td>
             {% endfor %}
         {% endfor %}
     </tr>
 
-    {# Row 5 - role #}
+    {# FIFTH ROW = compound role #}
     <tr>
-        {% for key,input in reaction.inputs|sort_addition_order %}
-            {% for compound in input.components %}
-                <td>{{ compound|compound_role }}</td>
+        {% for key, input in reaction.inputs|sort_addition_order %}
+            {% for compound, border in input.components|get_input_borders %}
+                <td class="{{ border }}">{{ compound|compound_role }}</td>
             {% endfor %}
         {% endfor %}
-        <td></td>
+    </tr>
+
+    {# LAST ROW = addition order groups #}
+    <tr><td class="clean"><br></td></tr>
```

Comment:
Do we need to wrap this in an `if` so that it doesn't get displayed if the addition order is not defined?

Reply:
Done.